### PR TITLE
cpu/stm32/periph/usbdev_fs: add support for STM32F3 family

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR provides some changes of the `cpu/stm32/periph/usbdev_fs` driver to support STM32F3. In detail, it includes the following changes:
- Small bug fix of the register name (commit 1ed26d83a20fee62b52297682eb781c378d59b27)
- Conditional conpilation of the Clock Recovery System configuration (commit b287a72c7afb8d950abda986b3f67a099a354bfd)
- Defiinition of GPIOs for USB_DM and USB_DP signals as additional function instead of alternative function (commit 332c39908e63b9787425a1e8154a7d8e9d5f1a7d)
- Definition of a GPIO used for USB Connect/Disconnect (commit 3693e16b770cc5f57f6f238f4b1ff028c7ca0713)
- Changes to support STM32F3 family (commit ca9cedb2ac508111c4a2457da10844af88be35b1)
- `periph_usbdev` support for `nucleo-f303ze`

Other boards using STM32F303 MCU such as `stm32f3discovery` should also work with these changes but I don't have the hardware at hand. It should also be possible to enable the `periph_usbdev` support using this driver for other boards using other STM32 families such as `b-l072z-lrwan1` or `lsn50`.

### Testing procedure

1. Compile and flash
  ```
  BOARD=nucleo-f303ze make -C tests/usbus_hid/ flash term
  ```
  The test should work as expected.
2. Compile and flash
  ```
  BOARD=p-nucleo-wb55 make -C tests/usbus_hid/ flash term
  ```
  The test should still work for this board which was the only one that was using this `periph_usbdev` driver before.

### CODE REVIEW

File: boards/nucleo-f303ze/Kconfig
Comment (benpicco) [inline boards/nucleo-f303ze/Kconfig:25]:
```suggestion
```
Wrong PR :wink:

File: boards/nucleo-f303ze/Makefile.features
Comment (benpicco) [inline boards/nucleo-f303ze/Makefile.features:15]:
```suggestion
```

File: cpu/stm32/periph/usbdev_fs.c
Comment (benpicco) [inline cpu/stm32/periph/usbdev_fs.c:198]: Want to fix that while you're at it?
```suggestion
    while (!(CRS->ISR & CRS_ISR_ESYNCF)) {}
```

**Review by benpicco — APPROVED** [review]
This looks all very straightforward - I should give #17812 another try based on this.

All good from my side, I trust your testing - but some feature has sneaked in that is unknown to Kconfig yet :wink:

File: boards/nucleo-f303ze/Kconfig
Comment (gschorcht) [reply boards/nucleo-f303ze/Kconfig:25]: This is intentional. PR #18592 is already in a state where it can be merged immediately. We are just waiting for a final CI compilation. So instead of re-basing after merging PR #18592 and adding this line afterwards, I already added this line so I wouldn't forget. OK, I should have added the label `State: wait for other PR` or at least a comment that this PR depends on PR #18592, even though it actually doesn`t.

Translated with www.DeepL.com/Translator (free version)

Comment (gschorcht) [reply boards/nucleo-f303ze/Kconfig:25]: Once PR #18592 is merged, I will rebase it.

**Comment by gschorcht** [comment]
> This looks all very straightforward - I should give #17812 another try based on this.

I have rebased this PR so that you could give PR #17812 a try on top of this PR.

**Comment by benpicco** [comment]
Looks like tinyUSB is still causing trouble with this PR

File: boards/nucleo-f303ze/Kconfig
Comment (benpicco) [inline boards/nucleo-f303ze/Kconfig:25]: Now this has to be removed after all 

```suggestion
```

Comment (gschorcht) [reply boards/nucleo-f303ze/Kconfig:25]: Oops, I shouldn't push changes while sitting in a meeting :worried:

Comment (benpicco) [reply boards/nucleo-f303ze/Kconfig]: That depends entirely on the meeting :smile:

**Comment by gschorcht** [comment]
> Looks like tinyUSB is still causing trouble with this PR

Ok, I see. Indeed, the tinyUSB feature requires some changes for this board. Even though I already have these changes and a number of other changes working locally, I will have to provide them one by one as PR. Therefore, I removed the feature for now.

File: cpu/stm32/periph/usbdev_fs.c
Comment (dylad) [inline cpu/stm32/periph/usbdev_fs.c:220]: `gpio_set()` is used both for attach and detach operations. Is this intentional ?

Comment (gschorcht) [reply cpu/stm32/periph/usbdev_fs.c:220]: Good catch :sunglasses:

File: cpu/stm32/periph/usbdev_fs.c
Comment (dylad) [inline cpu/stm32/periph/usbdev_fs.c:202]: Isn't `gpio_set()` for attach and `gpio_clear()` for detach ?

Comment (gschorcht) [reply cpu/stm32/periph/usbdev_fs.c]: I had obviously already drunk too many beers last night at the semester opening party :wink:

**Comment by dylad** [comment]
Looks good to me. Changes make sense.
Could you please share with us the output of any usbdev test with a STM32F3 ? I don't have any of them unfortunately.
Otherwise you can also squash since my comment is addressed and @benpicco was also happy.

**Comment by gschorcht** [comment]
> Could you please share with us the output of any usbdev test with a STM32F3 ?
```python
Oct  5 10:07:50 gunny8 kernel: [2063262.508628] usb 1-4.2.2.4: reset high-speed USB device number 41 using xhci_hcd
Oct  5 10:07:53 gunny8 kernel: [2063265.148365] usb 1-2: new full-speed USB device number 50 using xhci_hcd
Oct  5 10:07:53 gunny8 kernel: [2063265.305844] usb 1-2: New USB device found, idVendor=1209, idProduct=7d01, bcdDevice= 1.00
Oct  5 10:07:53 gunny8 kernel: [2063265.305847] usb 1-2: New USB device strings: Mfr=3, Product=2, SerialNumber=4
Oct  5 10:07:53 gunny8 kernel: [2063265.305849] usb 1-2: Product: nucleo-f303ze
Oct  5 10:07:53 gunny8 kernel: [2063265.305850] usb 1-2: Manufacturer: RIOT-os.org
Oct  5 10:07:53 gunny8 kernel: [2063265.305851] usb 1-2: SerialNumber: A5D5032437D0B7D8
Oct  5 10:07:53 gunny8 kernel: [2063265.309703] hid-generic 0003:1209:7D01.0168: hiddev2,hidraw7: USB HID v1.10 Device [RIOT-os.org nucleo-f303ze] on usb-0000:00:14.0-2/input0
```
```shell
echo "hello" > /dev/hidraw7
```
```shell
--- Miniterm on /dev/ttyACM0  115200,8,N,1 ---
--- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H ---
main(): This is RIOT! (Version: 2022.10-devel-1018-g08bd4-drivers/usbdev_stm32f3)
RIOT USB HID echo test
Write input to the hidraw device under /dev/hidrawX, and see if it gets echoed here
USB_HID rx_cb: Test argument 
Msg received via USB HID: 
hello
```

**Review by dylad — APPROVED** [review]
Perfect ! Let's merge this.

**Comment by gschorcht** [comment]
@benpicco @dylad Thanks for reviewing and merging.

**Pull request merged**
